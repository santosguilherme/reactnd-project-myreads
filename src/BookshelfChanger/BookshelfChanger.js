import React, {Component} from 'react';
import PropTypes from 'prop-types';

import shelves from '../commons/shelves';

import './BookshelfChanger.css';


class BookshelfChanger extends Component {
    handleChangeSelect = event => {
        const {onChangeOption} = this.props;
        onChangeOption(event.target.value);
    };

    render() {
        const {shelf} = this.props;

        return (
            <div className="book-shelf-changer">
                <select
                    value={shelf}
                    onChange={this.handleChangeSelect}
                >
                    <option value="none" disabled>Move to...</option>
                    {shelves.map(shelf => (
                        <option
                            key={shelf.type}
                            value={shelf.type}
                        >
                            {shelf.label}
                        </option>
                    ))}
                    <option value="none">None</option>
                </select>
            </div>
        );
    }
}

BookshelfChanger.defaultProps = {
    shelf: 'none'
};

BookshelfChanger.propTypes = {
    shelf: PropTypes.string,
    onChangeOption: PropTypes.func.isRequired
};

export default BookshelfChanger;