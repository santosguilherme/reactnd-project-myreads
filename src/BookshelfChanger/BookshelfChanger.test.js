import React from 'react';
import {shallow, mount} from 'enzyme';

import BookshelfChanger from './BookshelfChanger';

it('renders without crashing', () => {
    const onChangeOptionMock = jest.fn();

    shallow(<BookshelfChanger onChangeOption={onChangeOptionMock}/>);
});

it('Should call the onChangeOption property when change selected option', () => {
    const onChangeOptionMock = jest.fn();
    const test = mount(<BookshelfChanger onChangeOption={onChangeOptionMock}/>);

    test.find('select').simulate('change');
    expect(onChangeOptionMock).toHaveBeenCalledTimes(1);
});