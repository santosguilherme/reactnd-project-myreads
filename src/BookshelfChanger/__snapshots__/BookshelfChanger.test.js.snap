// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookshelfChanger
    onChangeOption={[Function]}
    shelf="none"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <select
        onChange={[Function]}
        value="none"
>
        <option
                disabled={true}
                value="none"
        >
                Move to...
        </option>
        <option
                value="currentlyReading"
        >
                Currently Reading
        </option>
        <option
                value="wantToRead"
        >
                Want to Read
        </option>
        <option
                value="read"
        >
                Read
        </option>
        <option
                value="none"
        >
                None
        </option>
</select>,
      "className": "book-shelf-changer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <option
            disabled={true}
            value="none"
>
            Move to...
</option>,
          Array [
            <option
              value="currentlyReading"
>
              Currently Reading
</option>,
            <option
              value="wantToRead"
>
              Want to Read
</option>,
            <option
              value="read"
>
              Read
</option>,
          ],
          <option
            value="none"
>
            None
</option>,
        ],
        "onChange": [Function],
        "value": "none",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Move to...",
            "disabled": true,
            "value": "none",
          },
          "ref": null,
          "rendered": "Move to...",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "currentlyReading",
          "nodeType": "host",
          "props": Object {
            "children": "Currently Reading",
            "value": "currentlyReading",
          },
          "ref": null,
          "rendered": "Currently Reading",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "wantToRead",
          "nodeType": "host",
          "props": Object {
            "children": "Want to Read",
            "value": "wantToRead",
          },
          "ref": null,
          "rendered": "Want to Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": "read",
          "nodeType": "host",
          "props": Object {
            "children": "Read",
            "value": "read",
          },
          "ref": null,
          "rendered": "Read",
          "type": "option",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "None",
            "value": "none",
          },
          "ref": null,
          "rendered": "None",
          "type": "option",
        },
      ],
      "type": "select",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <select
          onChange={[Function]}
          value="none"
>
          <option
                    disabled={true}
                    value="none"
          >
                    Move to...
          </option>
          <option
                    value="currentlyReading"
          >
                    Currently Reading
          </option>
          <option
                    value="wantToRead"
          >
                    Want to Read
          </option>
          <option
                    value="read"
          >
                    Read
          </option>
          <option
                    value="none"
          >
                    None
          </option>
</select>,
        "className": "book-shelf-changer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <option
              disabled={true}
              value="none"
>
              Move to...
</option>,
            Array [
              <option
                value="currentlyReading"
>
                Currently Reading
</option>,
              <option
                value="wantToRead"
>
                Want to Read
</option>,
              <option
                value="read"
>
                Read
</option>,
            ],
            <option
              value="none"
>
              None
</option>,
          ],
          "onChange": [Function],
          "value": "none",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Move to...",
              "disabled": true,
              "value": "none",
            },
            "ref": null,
            "rendered": "Move to...",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "currentlyReading",
            "nodeType": "host",
            "props": Object {
              "children": "Currently Reading",
              "value": "currentlyReading",
            },
            "ref": null,
            "rendered": "Currently Reading",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "wantToRead",
            "nodeType": "host",
            "props": Object {
              "children": "Want to Read",
              "value": "wantToRead",
            },
            "ref": null,
            "rendered": "Want to Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": "read",
            "nodeType": "host",
            "props": Object {
              "children": "Read",
              "value": "read",
            },
            "ref": null,
            "rendered": "Read",
            "type": "option",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "None",
              "value": "none",
            },
            "ref": null,
            "rendered": "None",
            "type": "option",
          },
        ],
        "type": "select",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
